### ISSUE TITLE
staticcheck: check for calling regexp.Match() (etc) in a loop

### ISSUE THREAD

**Issue by dgryski** [dominikh/go-tools#1]
Given how expensive it it vs. calling regexp.Compile() first and then re.Match(), this seems like an obvious mistake to flag.

For example, https://play.golang.org/p/49sgROkOS1 (takes too long for playground).

**Comment by shurcooL** [comment]
I like this in general, but have a concern. What if the loop runs only 2 or 3 times and it's not in a hot path?

**Comment by dominikh** [comment]
> What if the loop runs only 2 or 3 times and it's not in a hot path?

Once I've implemented the check, I will run it against my corpus and look for unhelpful suggestions. If there is a significant amount of them, one possible approach would be relying on VRP information to estimate the number of iterations. Of course that would only estimate the loop itself, not the surrounding context. But we can tackle these issues after we got some actual metrics.
